akka-docs/src/main/paradox/typed/persistence-active-active.md

```diff
@@ -0,0 +1,191 @@
+
+Each replica can have a separate database, e.g. different SQL databases in each DC, in that case provide a different query plugin id per replica:
+
+Scala
+:  @@snip [ActiveActiveCompileOnlySpec.scala](/akka-persistence-typed-tests/src/test/scala/docs/akka/persistence/typed/ActiveActiveCompileOnlySpec.scala) { #factory }
+
+In this scenario each replica reads from each other's database effectively providing cross region replication for any database that has an Akka Persistence plugin. 
+
+The function passed to both factory methods return an `EventSourcedBehavior` and provide access to the @api[ActiveActiveContext] that has the following methods:
+
+* entityId 
+* replicaId
+* allReplicas
+* persistenceId - to provide to the `EventSourcedBehavior` factory. This **must be used**.
+
+As well as methods that **can only be** used in the event handler. The values these methods return relate to the event that is being processed.
+
+* origin: The ReplicaId that originally created the event
+* concurrent: Whether the event was concurrent with another event as in the second diagram above
+* recoveryRunning: Whether a recovery is running. Can be used to send commands back to self for side effects that should only happen once.
+* currentTimeMillis: similar to `System.currentTimeMillis` but guaranteed never to go backwards
+
+The factory returns a `Behavior` that can be spawned like any other behavior.
+
+## Resolving conflicting updates
+
+### Conflict free replicated data types
+
+TODO example once CRDTs are in
+
+### Last writer wins
+
+Sometimes it is enough to use timestamps to decide which update should win. Such approach relies on synchronized clocks, and clocks of different machines will always be slightly out of sync. Timestamps should therefore only be used used when the choice of value is not important for concurrent updates occurring within the clock skew.
+ 
+ In general, last writer wins means that the event is used if the timestamp of the event is later (higher) than the timestamp of previous local update, otherwise the event is discarded. There is no built-in support for last writer wins, because it must often be combined with more application specific aspects.
+ 
+![images/lww.png](images/lww.png)
+
+There is a small utility class @api[LwwTime] that can be useful for implementing last writer wins semantics.
+It contains a timestamp representing current time when the event was persisted and an identifier of the
+data center that persisted it. When comparing two @api[LwwTime] the greatest timestamp wins. The replica
+identifier is used if the two timestamps are equal, and then the one from the data center sorted first in
+alphanumeric order wins.
+
+The nature of last writer wins means that if you only have one timestamp for the state the events must represent an
+update of the full state, otherwise there is a risk that the state in different data centers will be different and
+not eventually converge to the same state.
```

Comment:
I'm finding this sentence confusing. This needs to be re-phrased or maybe a comma is missing: 

_"if you only have one timestamp for the state the events must represent an update of the full state". Not clear to what._ 


We are also repeating 'the state' few times. For instance, the last part could omit it. 

_"..otherwise there is a risk that the state in different data centers will be different and will not eventually converge."_

Reply:
Removed the final state

Reply:
Any ideas to make the sentence more clear? It is then described via the example.

Reply:
I will think on something.